### ISSUE TITLE
Thunderbird is failing autodiscovering Address Books

### ISSUE THREAD

**Issue by roughnecks** [Kozea/Radicale#1543]
Hello,
after having successfully set up calendars auto-discover in TB, I noticed that address books can be imported only if specifying the full URL provided by Radicale's webpage.

When adding an email account in TB it should support discovering both remote calendars and address books, since it tells you so, but even trying manually to add a CardDAV A.book, specifying only the dav.domain.tld, and not the full A.book URL, it fails and tells this:

https://i.imgur.com/GINEjQr.jpeg

`No address books could be found to be imported`

To be noted: I'm trying this, logging in with just username and not "email"

https://termbin.woodpeckersnest.space/1rkrpq

I'm running Radicale from Debian 12, not sure who the fault is.. Radicale or Thunderbird, since calendars work after the small patch reported [here](https://github.com/Kozea/Radicale/issues/1536#issuecomment-2233346471)

**Comment by pbiering** [comment]
Hmm, works with
- Calendar: Thunderbird's internal implementation
- Addressbooks: Thunderbird in combination with CardBook Add-on
both shows available calendars or addressbooks and let me choose what to subscribe

**Comment by roughnecks** [comment]
But TB now has native support for both, why using an extension?

**Comment by roughnecks** [comment]
> But TB now has native support for both, why using an extension?

Oh, misread your post. Well, there's something wrong with my setup then, because it doesn't work with or without the extension.

**Comment by jonassmedegaard** [comment]
Try check that your front-end web server properly handles meta-requests for caldav and carddav.

One way to do that is with the following from a command-line (replacing USERNAME and CALENDARHOST to local values:

```shell
wget -u USERNAME https://CALENDARHOST/.well-known/carddav/
```

**Comment by jonassmedegaard** [comment]
whops, I meant curl, not wget.

For wget use options `--user --ask-password --spider`.

**Comment by roughnecks** [comment]
```
wget --user simone --ask-password --spider https://dav.example.tld/.well-known/carddav/
Password for user ‘simone’:
Spider mode enabled. Check if remote file exists.
--2024-07-17 20:39:13--  https://dav.example.tld/.well-known/carddav/
Resolving dav.example.tld (dav.example.tld)... IP-Redacted, IP-Redacted
Connecting to dav.example.tld (dav.example.tld)|IP-Redacted|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://dav.example.tld/radicale/ [following]
Spider mode enabled. Check if remote file exists.
--2024-07-17 20:39:13--  https://dav.example.tld/radicale/
Connecting to dav.example.tld (dav.example.tld)|IP-Redacted|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: /radicale/.web [following]
Spider mode enabled. Check if remote file exists.
--2024-07-17 20:39:13--  https://dav.example.tld/radicale/.web
Connecting to dav.example.tld (dav.example.tld)|IP-Redacted|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: /radicale/.web/ [following]
Spider mode enabled. Check if remote file exists.
--2024-07-17 20:39:13--  https://dav.example.tld/radicale/.web/
Connecting to dav.example.tld (dav.example.tld)|IP-Redacted|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5704 (5.6K) [text/html]
Remote file exists and could contain further links,
but recursion is disabled -- not retrieving.
```

**Comment by roughnecks** [comment]
https://github.com/Kozea/Radicale/issues/1160

**Comment by roughnecks** [comment]
Just to be more specific, I have these configurations in my base domain `https://domain.tld` (not the dav. subdomain)

```
   location /mail {
        try_files /.well-known/autoconfig/mail/config-v1.1.xml =404;
   }

   location /.well-known/caldav {
        return 301 https://dav.domain.tld;
   }

   location /.well-known/carddav {
        return 301 https://dav.domain.tld;
   }
   ```

Mail accounts are automatically configured by TB, calendars also work now after this patch https://github.com/Kozea/Radicale/issues/1536#issuecomment-2233346471
so the only thing left are address books.

I have a shared VPS and I would like for users to be able to configure their mail account in TB and get everything set up at once - mail, calendars and address books.

**Comment by jonassmedegaard** [comment]
> #1160

Yes, Radicale does not handle those hints. That's the reason I suggested you to try check that your **front-end** web server handles it.

**Comment by roughnecks** [comment]
Ok, then I pasted what you wanted to see, right?

**Comment by jonassmedegaard** [comment]
I guess - it seems you shared a mangled output with some dummy domains, where the use or not of subdomain "dav" seemingly does not correlate with your own described setup.
But really I didn't mean for you to show it to me, only suggested a check that you could do on your own, so if you know what you mangled and how it is corresponds with your setup, then you've succesfully checked your setup and can move on to other causes of failure.

**Comment by roughnecks** [comment]
I just replaced "myrealdomain.tld" with "example.tld", so when you see "dav.example.tld" that is "dav.myrealdomain.tld"

meaning: I haven't mangled anything else but the base domain name and tld.

Anyway, thanks for your help.

**Comment by roughnecks** [comment]
### Calendars: dav.woodpeckersnest.space

![image](https://github.com/user-attachments/assets/aed34ca4-5134-42cc-9426-dd7011d2719f)
![image](https://github.com/user-attachments/assets/7670faae-a87f-431e-a4fb-06a426887c04)
![image](https://github.com/user-attachments/assets/9d28a7c0-bce5-4275-a882-fbc40e89143d)

### Calendars: woodpeckersnest.space

![image](https://github.com/user-attachments/assets/0808033a-e10b-4b81-8e85-e8bc6f95354b)
![image](https://github.com/user-attachments/assets/4bc0df9c-613c-4231-ae99-aae53883dae5)
![image](https://github.com/user-attachments/assets/80c14ab9-8cc5-4617-97c1-c63960ba4e7c)


### Address Book: dav.woodpeckersnest.space

![image](https://github.com/user-attachments/assets/5987d6b4-7308-4d32-a25d-2d482f4bd0f1)
![image](https://github.com/user-attachments/assets/984c2988-84b9-4dec-8f66-03b963a0e7e6)
(no address books were found to be added for specified account)

### Address Book: woodpeckersnest.space

![image](https://github.com/user-attachments/assets/840fb43d-e668-426d-bfb0-7030d047c94d)
![image](https://github.com/user-attachments/assets/22636380-0284-4a7d-81bf-8f77c14a55df)
(connection failed)

What I mean is, I have the same setup for caldav and carddav, so why the difference?
Now I posted my domain, so everything is more clear... `woodpeckersnest.eu` should be the same as `.space`, I own both of them.

**Comment by jonassmedegaard** [comment]
Perhaps helpful to examine logfiles of your front-end web server, to understand exactly what your Thunderbird client is requesting in each test case, and try derive a pattern from that.

**Comment by itglob** [comment]
this is because Radicale terminate all DAV request to url like /username/

for example baikal server response /dav.php/principals/username/
 to current-user-principal and /dav.php/addressbooks/username/ to addressbook-home-set


but thunderbird has some "optimization" in file /thunderbird-115.13.0/comm/mailnews/addrbook/modules/CardDAVUtils.jsm

```
let triedURLs = new Set();
async function tryURL(url) {
if (triedURLs.has(url)) {
return;
}
triedURLs.add(url);
```
